src/index.ts

```diff
@@ -209,4 +210,49 @@ app.post("/attendance", async (c) => {
   return new Response();
 });
 
+app.get("/attendances/:course_id", async(c) => {
+  const subjectID = c.req.param('course_id');
+  const honoSession = c.get('session');
+  const session = honoSession.get("login") as Session | null;
+  if ( !session ){
+    return c.text("Unauthorized", 401);
+  }
+
+  //科目が存在するか、履修しているかをチェック
+  const [subjectResult, registration] = c.env.DB.batch([
+    c.env.DB.prepare("SELECT id FROM subjet WHERE id = ?").bind(subjectID),
+    c.env.DB.prepare("SELECT * registration WHERE subject_id = ?1 AND student_id = ?2").bind(subjectID, session.account.id)
+  ])
+  const subjectExists = subjectResult.results.length === 1;
+  const isRegistration = registration.results.length === 1;
+  if ( !subjectExists ){
+    return c.text("Not Found", 404);
+  }
+  if ( !isRegistration ){
+    return c.text("Not Found", 404);
+  }
+
+  /*出席状況を取得*/
+  const { on_time, late, miss } = c.env.DB
```

Comment:
```suggestion
  const entry = await c.env.DB
```
`Promise` が `await` されていないので分割代入で取り出せません.
また, クエリが不正だった場合にそのエラーを特定するのが難しくなるので, `entry` などの変数に一旦受け取っておいてから, 以下のように問題無いか検査して取り出すコードを書いたほうがよさそうです.
```ts
  if (entry === null) {
    throw new Error("SUM up query was invalid");
  }
  const { on_time, late, miss } = entry;
```